Carting/src/WebApi/Controllers/V2/CartController.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Carting.WebApi.Application.Carts.Commands.AddItemToCart;
+using Carting.WebApi.Application.Carts.Commands.RemoveItemFromCartCommand;
+using Carting.WebApi.Application.Carts.Queries.GetCart;
+using Carting.WebApi.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Carting.WebApi.Controllers.V2;
+
+[ApiVersion("2")]
+public class CartsController : ApiControllerBase
+{
+    /// <summary>
+    /// Returns a specific cart and its associated items
+    /// </summary>
+    /// <param name="query"></param>
+    /// <returns></returns>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     GET /
+    ///     {
+    ///        "CartId": 1
+    ///     }
+    ///
+    /// </remarks>
+    /// <response code="200">Returns the cart</response>
+    /// <response code="404">If the cart does not exist</response>
+    /// <response code="500">Server error</response>
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpGet]
+    public async Task<ActionResult<IList<CartItemDto>>> GetCart([FromQuery] GetCartQuery query)
+    {
+        var cart = await Mediator.Send(query);
+        return Ok(cart.Items);
+    }
+
+    /// <summary>
+    /// Adds an item to a specific cart. If there was no cart for the specifiedkey, it creates the cart as well.
+    /// </summary>
+    /// <param name="cartId"></param>
+    /// <param name="command"></param>
+    /// <returns></returns>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     POST /some-id/Items
+    ///     {
+    ///        "Id": 1,
+    ///        "Name": "Some item name",
+    ///        "CurrencyCode": "EUR",
+    ///        "Price":5.99,
+    ///        "Quantity":2,
+    ///        "WebImage": {
+    ///           "Uri": "http://localhost/some-image-url.png",
+    ///           "AltText": "Missing image"
+    ///         }
+    ///     }
+    ///
+    /// </remarks>
+    /// <response code="201">Item created</response>
+    /// <response code="422">Request couldn't be processed - item is already added</response>
+    /// <response code="500">Server error</response>
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [Route("{cartId}/Items")]
+    [HttpPost]
+    public async Task<ActionResult> AddItemToCart(string cartId, AddItemToCartCommandDto command)
```

Comment:
If I understand correctly now you have two API versions and only one difference between them. However all endpoints instantly got v2. This is complicated and unnecessary, as you would like to release endpoints in stages.

Reply:
I'm not sure if I misunderstood it, but the task on the learn page states the following:

> Version 2 – the same as Version 1 but with the following changes:
>  Get cart info.
> Returns a list of cart items instead of cart model.

This tells me that V2 should contain the unchanged endpoints of the V1 controller as well. Although, I completely agree with you.